R/dfm-methods.R

```diff
@@ -293,5 +292,5 @@ NULL
 #' @export
 #' @seealso [Comparison] operators
 setMethod("Compare", c("dfm", "numeric"), function(e1, e2) {
-    as(callGeneric(as(e1, "dgCMatrix"), e2), "lgCMatrix")
+    as(callGeneric(as(as(as(e1, "CsparseMatrix"), "generalMatrix"), "dMatrix"), e2), "CsparseMatrix")
```

Comment:
I wonder why you need to apply as, because DFM is always a dgCMatrix. Do you need to strip the DFM class to call the generic or something?

Reply:
Yes we do, but it could be accomplished more simply, so I changed it to 
```r
setMethod("Compare", c("dfm", "numeric"), function(e1, e2) {
    as(callGeneric(as(e1, "CsparseMatrix"), e2), "CsparseMatrix")
})
```

also I updated the documentation to reflect that it always returns an lgCMatrix.